app/src/main/java/org/wikipedia/categories/CategoryActivityViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package org.wikipedia.categories
+
+import android.os.Bundle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.paging.*
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.page.PageTitle
+
+class CategoryActivityViewModel(bundle: Bundle) : ViewModel() {
+
+    val pageTitle = bundle.getParcelable<PageTitle>(CategoryActivity.EXTRA_TITLE)!!
+
+    val categoryMembersFlow = Pager(PagingConfig(pageSize = 10)) {
+        CategoryMembersPagingSource(pageTitle, "page")
+    }.flow.cachedIn(viewModelScope)
+
+    val subcategoriesFlow = Pager(PagingConfig(pageSize = 10)) {
+        CategoryMembersPagingSource(pageTitle, "subcat")
+    }.flow.cachedIn(viewModelScope)
+
+    class CategoryMembersPagingSource(
+            val pageTitle: PageTitle,
+            private val resultType: String
+    ) : PagingSource<String, PageTitle>() {
+        override suspend fun load(params: LoadParams<String>): LoadResult<String, PageTitle> {
+            return try {
+                val response = ServiceFactory.get(WikiSite.forLanguageCode(pageTitle.wikiSite.languageCode))
+                        .getCategoryMembers(pageTitle.prefixedText, resultType, params.loadSize, params.key)
+                val titles = response.query!!.pages!!.map { page ->
+                    PageTitle(page.title, pageTitle.wikiSite).also {
+                        it.description = page.description.orEmpty()
+                        it.thumbUrl = page.thumbUrl()
+                    }
```

Comment:
Looks like it will cause a language variant issue here if we don't set the correct display title.

Would it be possible to call `page/summary` to get the correct `displaytitle`?

Reply:
I have updated API call to also return variant titles, similarly to how we do it in Search.